packages/jest/src/resultsProcessor.ts

```diff
@@ -0,0 +1,108 @@
+ *  are extracted out into their own test suites and results
+ */
+function modifyTestSuiteResults(testSuite: TestResult, testResult: AssertionResult) {
+    // Don't report the failure twice
+    testResult.status = 'disabled';
+    // Suites with only a11y errors should be marked as passed
+    testSuite.numFailingTests -= 1;
+    // TODO (fix): Remove sa11y msg from test suite message
+    //  ANSI codes and test names in suite message makes it difficult
+    // testResult.failureMessages = [];
+    // testResult.failureDetails = [];
+    // testSuite.failureMessage = '';
+}
+
+/**
+ * Convert a11y violations from given Jest tests to A11y Test Results
+ */
+function convertA11yTestResult(testSuite: TestResult, testResult: AssertionResult, violations: AxeResults) {
+    const suiteName = testSuite.testFilePath.substring(testSuite.testFilePath.lastIndexOf('/') + 1);
+
+    modifyTestSuiteResults(testSuite, testResult);
+
+    violations.forEach((violation) => {
+        const wcagMetaData = new WcagMetadata(violation.tags).toString();
+        violation.nodes.forEach((a11yError) => {
+            // TODO (refactor): Extract common code and reuse in regular a11y formatted error output
+            const suiteKey = `[Sa11y ${wcagMetaData} ${violation.id}]: ${suiteName}`;
+            if (!consolidatedErrors.has(suiteKey)) consolidatedErrors.set(suiteKey, []);
+            consolidatedErrors.get(suiteKey)?.push({
+                ...testResult,
+                fullName: `${a11yError.target[0]}`, // First CSS Selector
+                failureMessages: [
+                    `${errMsgHeader}: ${violation.help}
+ CSS Selectors: "${a11yError.target.join('; ')}"
+ Help: ${violation.helpUrl.split('?')[0]}
+ Tests: "${testResult.fullName}"`,
+                ],
+                failureDetails: [], // We don't need them anymore
+                ancestorTitles: [...new Set(testResult.ancestorTitles).add(testResult.fullName)], // Add all test's having the same a11y issue
+            } as AssertionResult);
+        });
+    });
+}
+
+/**
+ * Convert any a11y errors from test failures into their own test suite, results
+ */
+function processA11yErrors(testSuite: TestResult, testResult: AssertionResult) {
+    testResult.failureDetails.forEach((failure) => {
+        let error = (failure as FailureDetail).error;
+        // If using circus test runner https://github.com/facebook/jest/issues/11405#issuecomment-843549606
+        if (error === undefined) error = failure as A11yError;
+        if (error.name === A11yError.name) {
+            // TODO : What happens if there are ever multiple failureDetails?
+            //  Ideally there shouldn't be as test execution should be stopped on failure
+            convertA11yTestResult(testSuite, testResult, error.violations);
+        }
+    });
+}
+
+/**
+ * Custom results processor for a11y results. Only affects JSON results file output.
```

Comment:
@trevor-bliss I am replacing the Custom Jest reporter with a custom test results processor

* The Jest reporter was growing complex in trying to produce a JUnit XML compatible output - with duplication in representing/processing the XML struct.

* With the test results processor we have access to the test results before they are rendered to JSON (which is a different (untyped) format - not plain serialization) - hence it is more straightforward to process them. With this we don't have to duplicate effort in trying to produce a JUnit compatible output - we can reuse existing tools that convert Jest JSON to JUnit XML. I did run into the results processor earlier before working on the reporter but it is not as well documented as the reporter so I passed it up 🤦 

Working on cleaning up the code and hoping to have it up for review soon.

Reply:
Just to clarify the reporter is a uni-directional workflow where Jest hands the aggregated result output off to the reporter and doesn't expect anything to be returned back - leaving any results conversion, file handling etc completely to the reporter. This would be fine if we wanted to convert the results to a different format entirely but if we wanted to stick to Jest's JSON output schema, it becomes more work. We might be able to reuse https://github.com/facebook/jest/blob/master/packages/jest-test-result/src/formatTestResults.ts  but as you can see the structs behind the JSON output are not typed/defined explicitly - making it a bit trickier to test the conversion among other things.

Whereas the test processor sits in between - receiving the aggregated results from Jest and returning it back to Jest after processing - leaving further processing of results such as conversion to JSON, writing to file etc to Jest thereby reducing complexity a little and improving testability.